lib/nat/interface.go

```diff
@@ -21,7 +21,9 @@ const (
 
 type Device interface {
 	ID() string
-	GetLocalIPAddress() net.IP
+	GetLocalIPv4Address() net.IP
 	AddPortMapping(ctx context.Context, protocol Protocol, internalPort, externalPort int, description string, duration time.Duration) (int, error)
+	TryAddPinhole(ctx context.Context, protocol Protocol, port int, description string, duration time.Duration) (int, error)
```

Comment:
Does the `Try` prefix add something valuable? I don't think the other methods are guaranteed to succeed, either.

Reply:
You're right, the name isn't really concise. I've changed it now